Add CoC enforcement manual (#10651)
zephyrproject-rtos/zephyr: doc/code_of_conduct_enforcement_manual.rst

> .. _Zephyr_Code_of_Conduct_Enforcement_Manual

Reviewer: ```suggestion
.. _Zephyr_Code_of_Conduct_Enforcement_Manual:
```

> .. _Zephyr_Code_of_Conduct_Enforcement_Manual

Reviewer: All documents need to be included in the site's table of contents or the doc build will complain about orphan docs.

Code of Conduct Enforcement Manual
**********************************

> This is the enforcement manual followed by the Zephyr Projects’ Code of Conduct Working Group. It's used when we respond to an issue to make sure we're consistent and fair. It should be considered an internal document, but we're publishing it publicly in the interests of transparency.

Reviewer: Please keep line length < 80 characters when possible

> This is the enforcement manual followed by the Zephyr Projects’ Code of Conduct Working Group. It's used when we respond to an issue to make sure we're consistent and fair. It should be considered an internal document, but we're publishing it publicly in the interests of transparency.

Reviewer: ```suggestion
This is the enforcement manual followed by the Zephyr Project's Code of
Conduct Working Group. It's used when we respond to an issue to make
sure we're consistent, fair, and transparent in our procedures.
```

The Code of Conduct Working Group
*********************************

All responses to reports of conduct violations will be managed by a Code of Conduct Working Group ("the CoC WG").

> The Zephyr Project’s Governing Board ("the board") will establish this working group, comprised of at least five members. One member will be designated chair of the group and will be responsible for all reports back to the board. The board will review membership of the CoC WG on an annual basis. The CoC WG will consist of the project leadership (Governing Board Chair, TSC chair, Marketing Committee Chair, Community Representative and Project Director). 

Reviewer: Keep line length < 80 characters, and remove trailing spaces.

How the Code of Conduct Working Group Will Respond to Reports
*************************************************************

> When a report is sent to the committee they will immediately reply to the report to confirm receipt. The reporting email address is **report@zephyrproject.org**. This reply must be sent within 24 hours, and the group should strive to respond much quicker than that.

Reviewer: report@zephyrproject.org is not a working email address

See the reporting guidelines for details of what reports should contain. If a report doesn't contain enough information, the CoC WG will obtain all relevant data before acting. The CoC WG is empowered to act on the Zephyr Project’s behalf in contacting any individuals involved to get a more complete account of events.

The CoC WG will then review the incident and determine, to the best of their ability:

>  * What happened

Reviewer: need a blank line before a list, and don't indent the items:

    The CoC WG will then review the incident and determine, to the best of their ability:

    * What happened

Acting Unilaterally
*******************

If the act is ongoing (such as someone engaging in harassment in #zephyrproject), or involves a threat to anyone's safety (e.g. threats of violence), any CoC Working Group member may act immediately (before reaching consensus) to end the situation via a temporary ban while more information is gathered. In ongoing situations, any member may at their discretion employ any of the tools available to the working group, including bans and blocks.

If the incident involves physical danger, any member of the working group may -- and should -- act unilaterally to protect safety. This can include cooperating with law enforcement and speaking on behalf of the Zephyr Project. In situations where an individual group member acts unilaterally, they must report their actions to the working group for review within 24 hours.

Resolutions
***********

The working group must agree on a resolution by consensus. If the group cannot reach consensus and deadlocks for over a week, the group will turn the matter over to the Governing Board for resolution.

Possible responses may include:

>  * Taking no further action (if we determine no violation occurred).

Reviewer: Same here.  Add a blank line before the list and don't indent.

 * A request for a public or private apology. The chair will deliver this request. The group may, if it chooses, attach "strings" to this request: for example, the group may ask a violator to apologize in order to retain his or her membership on a mailing list.

Once a resolution is agreed upon, but before it is enacted, the working group will contact the original reporter and any other affected parties and explain the proposed resolution. The working group will ask if this resolution is acceptable, and must note feedback for the record. However, the working group is not required to act on this feedback.
Finally the working group will make a report for the Zephyr Governing Board (as well as the TSC team in the event of an ongoing resolution, such as a ban). The working group will never publicly discuss the issue; all public statements will be made by the Zephyr Project Governing Board.

Conflicts of Interest
*********************

In the event of any conflict of interest a working group member must immediately notify the other members, and recuse themselves if necessary.

> *Editor's note: Writing this document posed a unique challenge. Most similar guides are written on the assumption of an in-person event. However, the Zephyr community doesn't exist in one place, and most of the time we're spread out across the world and interact online. This makes trying to define and enforce community standards a different type of challenge. This document is adapted from the Django community, who adopted it from the Ada Initiative template and the PyCon 2013 Procedure for Handling Harassment Incidents, but changed to reflect the nature of our community. It is our expectation that this will be a living document and change as we grow to understand how to meet this challenge and best serve our community and ideals.*

Reviewer: Not sure we're that unique -- many open source projects have world-wide contributors, no?
